### PR TITLE
alternative cli implementation using prompt toolkit

### PR DESCRIPTION
**Pull request by chaudum**
this new cli implementation uses https://github.com/jonathanslenders/python-prompt-toolkit
additionally crash commands and output formatting were added

### CODE REVIEW

**Comment by dobe** [comment]
CHANGES are missing

File: src/crate/crash/command.py [hunk 1]
```diff
@@ -208,352 +208,254 @@ def pprint(self, rows, cols=None):
...
-        setattr(CrateCmd, "do_{0}".format(cmd_name.upper()), attr)
+    def pprint_json(self, obj):
+        json_str = json.dumps(obj, indent=2)
+        if self.is_tty:
+            return highlight(json_str, _json_lexer, _formatter)
+        return json_str
 
+    def pprint_csv(self, rows, cols):
+        csvfile = StringIO()
+        wr = csv.writer(csvfile, doublequote=False, escapechar='\\')
```
Comment (dobe) [inline src/crate/crash/command.py]: use stdout if not in tty, also for other printers

File: src/crate/crash/command.py [hunk 2]
```diff
@@ -112,448 +102,360 @@ class CrateCmd(Cmd):
...
+    def _quit(self, *args):
+        """ quit crash """
+        self.logger.warn(u'Bye!')
+        sys.exit(self.exit_code)
+
+    def _switch_format(self, fmt=None):
+        """ switch output format """
+        if fmt and fmt in self.OUTPUT_FORMATS:
+            self.output_format = fmt
+            return u'changed output format to {}'.format(fmt)
```
Comment (mfussenegger) [inline src/crate/crash/command.py]: maybe return the available formats here if it doesn't exist ?

**Comment by mfussenegger** [comment]
history support is missing

**Comment by mfussenegger** [comment]
ack

**Pull request merged**